sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageUtils.java

```diff
@@ -107,11 +111,19 @@ static Instant convertDotNetTicksToInstant(long dotNetTicks) {
      * @return The corresponding DeliveryState, or null if the disposition status is unknown.
      */
     public static DeliveryState getDeliveryState(DispositionStatus dispositionStatus, String deadLetterReason,
-        String deadLetterErrorDescription, Map<String, Object> propertiesToModify) {
+        String deadLetterErrorDescription, Map<String, Object> propertiesToModify,
+        ServiceBusTransactionContext transactionContext) {
         final DeliveryState state;
         switch (dispositionStatus) {
             case COMPLETED:
-                state = Accepted.getInstance();
+                if (transactionContext != null && transactionContext.getTransactionId() != null) {
```

Comment:
This check is repeated many times, you can simplify it by having a boolean before this switch method. `final boolean hasTransaction = context != null && ...`